src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -107,4 +107,37 @@ public int getTotalMessageCount() {
     return results.countEntities(FetchOptions.Builder.withLimit(1000));
   }
 
+  /** Returns the average length of all messages. */
+  public int getAverageMessageLength() {
+    Query query = new Query("Message");
+    PreparedQuery results = datastore.prepare(query);
+
+    int sum = 0;
+
+    for (Entity entity : results.asIterable()) {
+      sum = sum + ((String) entity.getProperty("text")).length();
+    }
+
+    return sum / getTotalMessageCount();
+
+  }
+
+  /**
+   * Returns the maximum length of all the messages. Returns -1 if there are no
+   * messages.
+   */
+  public int getMaxMessageLength() {
+    Query query = new Query("Message");
```

Comment:
It is always better to not have hardcoding done in the function.
Also, these are datastore column names. You should consider declaring global variables for the column name strings. This way you can re-use them at other places and in case if we decide to change the column name in future, we would not have to make column name change in the entire project, instead it would just a change at one place.